### PR TITLE
[Merged by Bors] - feat(ModelTheory): alternative statement of compactness

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/ModelTheory/Satisfiability.lean [hunk 1]
```diff
@@ -353,6 +353,24 @@ theorem ModelsBoundedFormula.realize_sentence {φ : L.Sentence} (h : T ⊨ᵇ φ
   exact Model.isSatisfiable M
 #align first_order.language.Theory.models_bounded_formula.realize_sentence FirstOrder.Language.Theory.ModelsBoundedFormula.realize_sentence
 
+/-- An alternative statement of the Compactness Theorem. A formula `φ` is modeled
```
Comment (awainverse) [inline Mathlib/ModelTheory/Satisfiability.lean:356]:
```suggestion
/-- An alternative statement of the Compactness Theorem. A formula `φ` is modeled by a
```

**Review by awainverse — COMMENTED** [review]
This is a totally reasonable characterization, and is some ways better than the definition I ended up choosing for finite satisfiability.
In fact, I've forgotten whether it would break things to just use this as the definition of finite satisfiability for the compactness theorem - perhaps that's worth trying.

**Comment by ChrisHughes24** [comment]
> This is a totally reasonable characterization, and is some ways better than the definition I ended up choosing for finite satisfiability. In fact, I've forgotten whether it would break things to just use this as the definition of finite satisfiability for the compactness theorem - perhaps that's worth trying.

I suspect for proving things like existence of really large models, you really prefer your characterization of compactness. I think we need both.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**